cmake/thirdparty/SetupMPI.cmake

```diff
@@ -33,6 +33,29 @@ set(_mpi_link_flags )
 
 message(STATUS "Enable FindMPI:  ${ENABLE_FIND_MPI}")
 
+#
+# TODO: should this be BLT_ENABLE_FIND_MPI?
+#
+if(ENABLE_FIND_MPI)
+    message(STATUS "FindMPI Enabled  (ENABLE_FIND_MPI == ON)")
+else()
+    message(STATUS "FindMPI Disabled (ENABLE_FIND_MPI == OFF) ")
+endif()
+
+
+if(BLT_USE_FIND_MPI_TARGETS)
+    message(STATUS "Using MPI CMake imported targets (BLT_USE_FIND_MPI_TARGETS == ON)")
+else()
+    message(STATUS "Not using MPI CMake imported targets (BLT_USE_FIND_MPI_TARGETS == OFF)")
+endif()
+
+
+
+if(NOT ENABLE_FIND_MPI AND BLT_USE_FIND_MPI_TARGETS)
```

Comment:
Seems like we could handle this better by converting `BLT_USE_FIND_MPI_TARGETS` to a `cmake_dependent_option`

Reply:
I tried `cmake_dependent_option`. It will just flip `BLT_USE_FIND_MPI_TARGETS` to OFF, when `ENABLE_FIND_MPI` is OFF. I think an explicit error is a better approach here --  better to error when things aren't consistent than to drive on to surprise town.